api/src/services/xml_generation/transformers/base_transformer.py

```diff
@@ -91,18 +102,37 @@ def _process_transform_rules(
                         )
                         continue
 
-                # Apply the transformation
-                transformed_value = self._apply_transform_rule(
-                    source_value, transform_rule, rule_config, current_path
-                )
+                # Apply the transformation (conditional transformations can handle None source values)
+                if transform_type == "conditional" or source_value is not None:
+                    transformed_value = self._apply_transform_rule(
+                        source_value, transform_rule, rule_config, current_path
+                    )
+                else:
+                    transformed_value = None
 
                 # Add to result if transformation succeeded and produced non-None value
                 if transformed_value is not None:
-                    target_field = transform_rule["target"]
-                    result[target_field] = transformed_value
-                    logger.debug(
-                        f"Transformed {'.'.join(current_path)} -> {target_field}: {source_value}"
-                    )
+                    # Handle one-to-many mappings that return dictionaries
+                    if isinstance(transformed_value, dict) and transform_type == "conditional":
+                        conditional_config = transform_rule.get("conditional_transform", {})
+                        if conditional_config.get("type") == "one_to_many":
+                            # Add all key-value pairs from one-to-many result
+                            result.update(transformed_value)
+                            logger.debug(
+                                f"One-to-many transform {'.'.join(current_path)} -> {list(transformed_value.keys())}"
+                            )
+                        else:
+                            target_field = transform_rule["target"]
+                            result[target_field] = transformed_value
+                            logger.debug(
+                                f"Transformed {'.'.join(current_path)} -> {target_field}: {source_value}"
+                            )
+                    else:
+                        target_field = transform_rule["target"]
+                        result[target_field] = transformed_value
+                        logger.debug(
+                            f"Transformed {'.'.join(current_path)} -> {target_field}: {source_value}"
+                        )
```

Comment:
I think these two elses are the same, any way to adjust the if-statements above to consolidate this a bit?